File: leetcode/3120/c/main.c
#define ALPHABET_COUNT 26

bool is_lowercase(char ch) {
    return ch >= 'a' && ch <= 'z';
}

bool is_uppercase(char ch) {
    return ch >= 'A' && ch <= 'Z';
}

int numberOfSpecialChars(char* word) {
    int cnt = 0;
    int n = strlen(word);
    int ch_map[ALPHABET_COUNT * 2] = {0};

    for (int i = 0; i < n; i++) {
        char ch = word[i];

        int idx = -1;

        if (is_lowercase(ch)) {
            idx = ch - 'a';
        } else if (is_uppercase(ch)) {
            idx = ch - 'A' + ALPHABET_COUNT;
        } else {
            printf("WTF????");
            exit(1);
        }

        ch_map[idx]++;
    }

    for (int i = 0; i < ALPHABET_COUNT; i++) {
        if (ch_map[i] > 0 && ch_map[i + ALPHABET_COUNT] > 0) {
            cnt++;
        }
    }

    return cnt;
}

File: leetcode/3120/rs/src/main.rs
struct Solution;
impl Solution {
    pub fn number_of_special_chars(word: String) -> i32 {
        const ALPHABET_COUNT: usize = (b'Z' - b'A' + 1) as usize;
        let mut cnt = 0;
        let mut ch_map = [0; ALPHABET_COUNT * 2];
        for &ch in word.as_bytes() {
            let idx = if ch.is_ascii_lowercase() {
                (ch - b'a') as usize
            } else if ch.is_ascii_uppercase() {
                (ch - b'A') as usize + ALPHABET_COUNT
            } else {
                panic!("WTF????");
            };
            ch_map[idx] += 1;
        }

        for i in 0..ALPHABET_COUNT {
            if ch_map[i] > 0 && ch_map[i + ALPHABET_COUNT] > 0 {
                cnt += 1;
            }
        }

        cnt
    }
}
fn main() {
    println!("Hello, world!");
}
